akka-persistence/src/main/scala/akka/persistence/serialization/MessageSerializer.scala

```diff
@@ -130,6 +130,7 @@ class MessageSerializer(val system: ExtendedActorSystem) extends BaseSerializer
 
     if (persistent.persistenceId != Undefined) builder.setPersistenceId(persistent.persistenceId)
     if (persistent.sender != Actor.noSender) builder.setSender(Serialization.serializedActorPath(persistent.sender))
+    if (persistent.manifest != PersistentRepr.Undefined) builder.setManifest(persistent.manifest)
```

Comment:
Yeah :< Wasn't caught by tests since they used the in-mem journal. I'll also want to enable the same tests using leveldb and then merge this.